Adds bluespace river away mission (spoilers for mission in the PR)

Anton-Kr opened the pull request:
:cl: isdatafamas
rscadd: Added arctic planet with science ship as an away mission
/:cl:

 
Credits to Maveriknight for writing the papers for the mission.
Credits to /vg/ for a lot of assets for this mission were ported from there. 
Every poor soul who helped me with code/git

This creates a two z-level away mission. The overground z-level features a science ship, that was exploring the second z-level which is underground. The planet is snowy and features mineable materials so the miners have stuff to do while the explorers all get murdered in the second z-level.

Second z-level contains a bluespace river. Touching the bluespace river or bumping into it results in instant death/ashing and intense levels of salt in deadchat. This will be known for anyone bothering to read the notes on the ship. It features alien artifacts consisting of an unlimited power cell and a blue colored decloner gun. It also features randomly generated alien big artifacts, which require some coordination to extract as there are only ladders available. Nothing too OP. The loot part of the cave is guarded by a ported from /vg/ hive defender. It will try to walk up to the target and activate it's attack mode. It will then get stuck in attack mode, remaining stationary, or will attack the target if it is in melee range. It's relatively sturdy, but shouldn't be a problem unless you backtrack yourself into the bluespace river.

Lore is pretty much up to the players to decide. 

Time2failtravis
Pictures
https://postimg.org/image/yjulp4ooz/
https://postimg.org/image/q1l5kspw3/
https://postimg.org/image/54oxg4zkz/

Anton-Kr committed:
Bluespace river initial commit

attempt numero 2

if travis could fuck off that'd be great thanks

BlueNexus commented:
Please don't use a unique bluespace sprite for this, we already have one that'll work just fine, and looks better. This one just looks weird and static, and is inconsistent with every other appearance of bluespace in the game.

chaoko99 commented:
You're only saying that because you made the sprite. It's a bluespace river, some kind of space leyline or something. We use the OTHER SPRITE when we are INSIDE bluespace, or it is encroaching upon us.

But I agree that the other one does not tile very well.

Anton-Kr commented:
The sprite is animated and has about 4 different variants. It looks fine in game. I dislike the original bluespace sprite, and I don't think it would be a good fit for this away.

PsiOmegaDelta commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,220 @@
+	icon_state = "hive_executioner_move"
+	flick("hive_executioner_movemode", src)
+
+	sleep(rand(transformation_delay_min, transformation_delay_max))
```
`sleep()` in a proc eventually called by a subsystem through `/Life()`.
Add `set waitfor = 0` at the top of the proc to prevent subsystem delays.

PsiOmegaDelta commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,220 @@
+	density = 1
+	temperature = 233
+
+/turf/unsimulated/wall/away/blueriver/livingwall/New()
```
``` 
/turf/unsimulated/wall/away/blueriver/livingwall/Initialize()
    . = ..()
   [etc.]
```

PsiOmegaDelta commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,220 @@
+	opacity = 0
+	dynamic_lighting = 0
+
+/turf/unsimulated/wall/supermatterriver/New()
```
``` 
/turf/unsimulated/wall/supermatterriver/Initialize()
    . = ..()
   [etc.]
```

PsiOmegaDelta commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,220 @@
+/turf/unsimulated/wall/supermatterriver/attack_ai(mob/user as mob)
+	user.examinate(src)
+
+#define MayConsume(A) (istype(A) && A.simulated && !isobserver(A))
```
Just a note: Observers are already covered by `A.simulated`

PsiOmegaDelta commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,220 @@
+
+
+
+/turf/unsimulated/wall/supermatterriver/Bumped(var/atom/movable/AM)
```
The code below and above look suspiciously copy-pasted from the bluespace rift code.

atlantiscze commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,220 @@
+#include "blueriver_areas.dm"
+/obj/effect/overmap/sector/arcticplanet
+	name = "Arctic Planet"
+	desc = "Sensor array an arctic planet with a small vessle on the surface. Scans further indicate strange energy levels below the planet's surface."
```
Weird wording and a typo. Probably supposed to be "Sensor arrays detect an arcitc planet with a small vessel on the surface. Further scans indicate strange energy levels from below the planet's surface."

Kearel commented:
I think it'd be interesting. Not sure how to fit the bluespace river in it all. Could just say, as you said, that there is a weak barrier between bluespace and reality there.

Anton-Kr commented:
@Kearel idea is to basically let the players decide, but yeah something like that. 
Also spoke with The Welp and he approved so, I guess can remove awaiting  lore tag.

comma commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,235 @@
+/turf/simulated/floor/away/blueriver/alienfloor/Initialize()
+	.=..()
+
+///turf/simulated/floor/away/blueriver/alienfloor/Initial()
```
ree random commented out stuff

Techhead0 commented on code/game/atoms.dm:
```diff
@@ -541,3 +541,7 @@ its easier to just keep the beam vertical.
 	else
 		return ..()
+
+/atom/proc/supermatter_act(atom/source, severity)
```
Why does this have a `severity` var? And why make something this simple a proc if nothing overrides it or does anything with it?

comma commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,235 @@
+/turf/unsimulated/wall/away/blueriver/livingwall/Initialize()
+	.=..()
+
+///turf/unsimulated/wall/away/blueriver/livingwall/Initial()
```
same

comma commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,235 @@
+/turf/unsimulated/wall/supermatterriver/Initialize()
+	.=..()
+
+///turf/unsimulated/wall/supermatterriver/Initial()
```
re

comma commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,235 @@
+		"<span class=\"warning\">You hear an unearthly noise.</span>")
+
+	playsound(src, 'sound/effects/supermatter.ogg', 50, 1)
+	qdel(user)
```
maybe supermatter_act here since you added it and all?

comma commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,163 @@
+
+
+/turf/simulated/floor/away/blueriver/alienfloor
+	name = "Glowing Floor"
```
lowercase turf names, would look odd in game messages

comma commented on maps/away/blueriver/blueriver.dm:
```diff
@@ -0,0 +1,163 @@
+		icon_state = "evilwall_[rand(1,8)]"
+
+/turf/unsimulated/wall/supermatter/no_spread
+	name = "Unknown"
```
lowercase, also maybe something like 'weird liquid'? Anything that it looks like on a glance

Anton-Kr commented:
aaaaaaaaaaaaaaaaaa why didnt this rebase work fuuuuuuck

Anton-Kr commented:
alright I cant really figure rebase out as it's literally giving me conflicts on other people's commits, on binary files. Anyone any ideas on what to do?

Anton-Kr committed:
# This is a combination of 2 commits.
# This is the 1st commit message:

Bluespace river added as away mission

# This is the commit message #2:

Automatic changelog generation for PR #19884 [ci skip]

Anton-Kr commented:
okay I give up.

chaoko99 commented:
Don't delete the branch, please.

Anton-Kr commented:
I'll remake this as a separate PR. Worry not.
